File: ios/ListenEventEmitter.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ListenEventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/ListenEventEmitter.swift
import Foundation

@objc(ListenEventEmitter)
class ListenEventEmitter: RCTEventEmitter {
  public static var shared: ListenEventEmitter!

  override init() {
    super.init()
    ListenEventEmitter.shared = self
  }

  override func supportedEvents() -> [String]! {
    ["RNListen.result", "RNListen.partial-result"]
  }
}
